lib/src/modules/apply.cc

```diff
@@ -0,0 +1,252 @@
+#include <pxp-agent/modules/apply.hpp>
+#include <pxp-agent/util/bolt_helpers.hpp>
+#include <pxp-agent/configuration.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/tokenizer.hpp>
+#include <leatherman/json_container/json_container.hpp>
+#include <leatherman/file_util/file.hpp>
+
+namespace fs = boost::filesystem;
+namespace lth_file = leatherman::file_util;
+namespace lth_jc = leatherman::json_container;
+
+namespace PXPAgent {
+namespace Modules {
+
+    static const std::string apply_ACTION { "apply" };
+    static const std::string apply_ACTION_INPUT_SCHEMA { R"(
+    {
+        "type": "object",
+        "properties": {
+            "environment": {
+                "type": "string"
+            },
+            "version": {
+                "type": "string"
+            },
+            "catalog": {
+                "type": "object"
+            },
```

Comment:
Would like to talk about this schema. See https://github.com/puppetlabs/orchestrator/pull/1298/

Reply:
done